collect_app/src/androidTest/java/org/odk/collect/android/support/CountingTaskExecutorIdlingResource.java

```diff
@@ -0,0 +1,34 @@
+package org.odk.collect.android.support;
+
+import androidx.arch.core.executor.testing.CountingTaskExecutorRule;
+import androidx.test.espresso.IdlingResource;
+
+public class CountingTaskExecutorIdlingResource implements IdlingResource {
```

Comment:
There is some good docs on `IdlingResource` [here](https://developer.android.com/training/testing/espresso/idling-resource). It seems Espresso is moving towards needing them more and more so it doesn't have to integrate with all the different scheduler mechanisms in Android (`Handler`, `ArchTaskExectuor`, coroutines and `WorkManager` to name a few).

Reply:
Did the person on the issue you filed about `WorkManager` and `Espresso` ever get back to you and was this the answer?